apps/dotcms-ui/src/app/view/components/dot-navigation/services/dot-navigation.service.ts

```diff
@@ -22,34 +22,71 @@ const replaceIdForNonMenuSection = (id) => {
     return replaceSectionsMap[id];
 };
 
-const setActiveItems = (id: string, collapsed: boolean) => (source: Observable<DotMenu[]>) => {
-    id = replaceIdForNonMenuSection(id) || id;
-
-    return source.pipe(
-        flatMap((menu: DotMenu[]) => menu),
-        map((menu: DotMenu) => {
-            setActiveUpdatedMenu(menu, id);
-            menu.isOpen = !collapsed && menu.active;
-
-            return menu;
-        }),
-        toArray()
-    );
-};
+interface DotActiveItemsProps {
+    id: string;
+    collapsed: boolean;
+    parent: string;
+}
 
-const setActiveUpdatedMenu = (menu: DotMenu, id: string) => {
-    let isActive = false;
+interface DotActiveItemsFromParentProps extends DotActiveItemsProps {
+    menus: DotMenu[];
+}
 
-    menu.menuItems.forEach((item: DotMenuItem) => {
-        if (item.id === id) {
-            item.active = true;
-            isActive = true;
-        } else {
-            item.active = false;
+function getActiveMenuFromParent({ menus, parent, collapsed, id }: DotActiveItemsFromParentProps) {
+    return menus.map((menu) => {
+        menu.active = false;
+
+        menu.menuItems = menu.menuItems.map((item) => ({
+            ...item,
+            active: false
+        }));
+
+        if (menu.id === parent) {
+            menu.active = true;
+            menu.isOpen = !collapsed && menu.active; // TODO: this menu.active what?
+            menu.menuItems = menu.menuItems.map((item) => ({
+                ...item,
+                active: item.id === id
+            }));
         }
+
+        return menu;
     });
+}
+
+const setActiveItems = ({ id, collapsed, parent }: DotActiveItemsProps) => (
+    source: Observable<DotMenu[]>
+) => {
+    id = replaceIdForNonMenuSection(id) || id;
 
-    menu.active = isActive;
+    return source.pipe(
+        map((m: DotMenu[]) => {
+            const menus: DotMenu[] = [...m];
+            let isActive = false;
+
+            if (parent) {
+                return getActiveMenuFromParent({ menus, parent, collapsed, id });
+            }
+
+            for (let i = 0; i < menus.length; i++) {
```

Comment:
Is there a way to avoid using nested for loops? Just asking, I was thinking but couldn't come up with any possible refactor.

Reply:
I thought about it, but the thing is that we have nested menus and we need to active the parent and the child.